cmd/kube-proxy/app/server_others_test.go

```diff
@@ -243,22 +246,80 @@ func Test_getLocalDetector(t *testing.T) {
 			errExpected: true,
 		},
 		{
-			mode:        proxyconfigapi.LocalMode("abcd"),
-			config:      &proxyconfigapi.KubeProxyConfiguration{ClusterCIDR: "10.0.0.0/14"},
+			mode:        proxyconfigapi.LocalModeClusterCIDR,
+			config:      &proxyconfigapi.KubeProxyConfiguration{ClusterCIDR: ""},
 			ipt:         utiliptablestest.NewFake(),
 			expected:    proxyutiliptables.NewNoOpLocalDetector(),
 			errExpected: false,
 		},
+		// LocalModeNodeCIDR
 		{
-			mode:        proxyconfigapi.LocalModeClusterCIDR,
+			mode:        proxyconfigapi.LocalModeNodeCIDR,
+			config:      &proxyconfigapi.KubeProxyConfiguration{ClusterCIDR: "10.0.0.0/14"},
+			ipt:         utiliptablestest.NewFake(),
+			expected:    resolveLocalDetector(t)(proxyutiliptables.NewDetectLocalByCIDR("10.0.0.0/24", utiliptablestest.NewFake())),
+			nodeInfo:    &v1.Node{Spec: v1.NodeSpec{PodCIDR: "10.0.0.0/24", PodCIDRs: []string{"10.0.0.0/24"}}},
```

Comment:
suggest a makeNode(cidr, cidrs) function to make this easier to read

Reply:
Ack.

Reply:
Done.